src/experiments.js

```diff
@@ -37,6 +42,9 @@ const COOKIE_EXPIRATION_INTERVAL = COOKIE_MAX_AGE_DAYS * 24 * 60 * 60 * 1000;
 /** @type {Object<string, boolean>} */
 let toggles_ = null;
 
+/** @type {!Promise} */
+const originTrialsPromise = enableExperimentsForOriginTrials(self);
```

Comment:
Any suggestions on how to get that to work with the async nature of the token parsing? `enableExperimentsForOriginTrials` needs to be able to return a promise.

I profiled `enableExperimentsFromToken` and at 10x throttling, Chomr reported about 5ms, though I'm dubious. Does that number sound reasonable to you?